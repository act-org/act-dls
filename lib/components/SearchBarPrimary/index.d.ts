/**
 * @prettier
 */
import * as React from 'react';
import { OutlinedInputProps } from '@material-ui/core';
export declare type Props = OutlinedInputProps;
declare const SearchBarPrimary: React.FC<Props>;
export default SearchBarPrimary;
//# sourceMappingURL=index.d.ts.map