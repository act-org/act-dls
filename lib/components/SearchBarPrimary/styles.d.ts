/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"&::-ms-clear, & *::-ms-clear" | "inputIconRoot" | "searchNotchedOutlined" | "searchRoot", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map