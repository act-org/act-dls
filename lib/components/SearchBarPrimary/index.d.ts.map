{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/SearchBarPrimary/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAG/B,OAA0B,EACxB,KAAK,IAAI,sBAAsB,EAChC,MAAM,gCAAgC,CAAC;AAKxC,oBAAY,KAAK,GAAG,sBAAsB,CAAC;AAE3C,QAAA,MAAM,gBAAgB,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CA2BrC,CAAC;AAEF,eAAe,gBAAgB,CAAC"}