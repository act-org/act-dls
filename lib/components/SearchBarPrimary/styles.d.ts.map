{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/SearchBarPrimary/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAKH,wBAyBG"}