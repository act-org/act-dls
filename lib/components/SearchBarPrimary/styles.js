"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styles = require("@material-ui/styles");

/**
 * @prettier
 */
var _default = (0, _styles.makeStyles)({
  '&::-ms-clear, & *::-ms-clear': {
    display: 'none'
  },
  inputIconRoot: {
    color: '#8f8f8f' // FIXME: COLORS.MEDIUM_GREY,

  },
  searchNotchedOutlined: {
    borderColor: '#f1f1f1'
  },
  searchRoot: {
    backgroundColor: '#f9f9f9',
    // FIXME: COLORS.LIGHT_GREY,
    borderRadius: 11,
    color: '#2d2d2d',
    // FIXME: COLORS.TEXT_PRIMARY,
    fontSize: 14,
    height: 40
  }
});

exports["default"] = _default;