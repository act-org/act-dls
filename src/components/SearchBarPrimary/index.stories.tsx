/**
 * @prettier
 */

/* eslint-disable sort-keys */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';

import SearchBarPrimary from '.';

export const Primary = (): React.ReactElement<any> => {
  const [value, setValue] = React.useState('');

  return (
    <SearchBarPrimary
      onChange={(e: any): void => {
        action('onChange()');

        setValue(e.target.value);
      }}
      placeholder={text('Placeholder', 'Search...')}
      value={value}
    />
  );
};

export default {
  component: SearchBarPrimary,
  parameters: {
    info: {
      text: 'This is the primary variant of the SearchBar component.',
    },
  },
  title: 'Molecules/SearchBar',
};
