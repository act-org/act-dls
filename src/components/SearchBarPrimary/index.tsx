/**
 * @prettier
 */

import * as React from 'react';

import InputAdornmentBase from '~/components/InputAdornmentBase';
import OutlinedInputBase, {
  Props as OutlinedInputBaseProps,
} from '~/components/OutlinedInputBase';
import { Magnify } from '~/icons';

import useStyles from './styles';

export type Props = OutlinedInputBaseProps;

const SearchBarPrimary: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => {
  const classes = useStyles();

  return (
    <OutlinedInputBase
      classes={{
        notchedOutline: classes.searchNotchedOutlined,
        root: classes.searchRoot,
      }}
      inputProps={{
        'aria-label': 'Search',
      }}
      startAdornment={
        <InputAdornmentBase
          classes={{
            root: classes.inputIconRoot,
          }}
          position="start"
        >
          <Magnify />
        </InputAdornmentBase>
      }
      {...props}
    />
  );
};

export default SearchBarPrimary;
