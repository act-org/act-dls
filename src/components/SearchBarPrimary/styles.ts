/**
 * @prettier
 */

import { makeStyles } from '@material-ui/styles';

export default makeStyles({
  '&::-ms-clear, & *::-ms-clear': {
    display: 'none',
  },
  inputIconRoot: {
    color: '#8f8f8f', // FIXME: COLORS.MEDIUM_GREY,
  },
  searchNotchedOutlined: {
    borderColor: '#f1f1f1',
  },
  searchRoot: {
    backgroundColor: '#f9f9f9', // FIXME: COLORS.LIGHT_GREY,
    borderRadius: 11,
    color: '#2d2d2d', // FIXME: COLORS.TEXT_PRIMARY,
    fontSize: 14,
    height: 40,
  },
});
