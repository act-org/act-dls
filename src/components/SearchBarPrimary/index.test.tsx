/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import SearchBarPrimary from '.';

describe('SearchBarPrimary', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <SearchBarPrimary onChange={noop} value="Value" />,
    );

    expect(container).toMatchSnapshot();
  });
});
