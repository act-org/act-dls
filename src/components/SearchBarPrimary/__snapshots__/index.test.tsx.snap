// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar matches the snapshot 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root makeStyles-searchRoot-4 MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
  >
    <div
      class="MuiInputAdornment-root makeStyles-inputIconRoot-2 MuiInputAdornment-positionStart"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
        />
      </svg>
    </div>
    <input
      aria-label="Search"
      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
      type="text"
      value="Value"
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline makeStyles-searchNotchedOutlined-3"
      style="padding-left: 8px;"
    >
      <legend
        class="PrivateNotchedOutline-legend-6"
        style="width: 0.01px;"
      >
        <span>
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
